contractcourt/chain_watcher.go

```diff
@@ -410,7 +410,7 @@ func (c *chainWatcher) closeObserver(spendNtfn *chainntnfs.SpendEvent) {
 		// actually greater than even one beyond our best known state
 		// number. This should ONLY happen in case we experienced some
 		// sort of data loss.
-		case broadcastStateNum > remoteStateNum+1:
+		case broadcastStateNum > remoteStateNum:
```

Comment:
Comments here and for the above case should be updated to reflect and describe the change. Would also be beneficial to comment on why the `+1` is not caught by the above case.